lib/embedded_ansible.rb

```diff
@@ -74,14 +69,54 @@ def self.services
   end
 
   def self.api_connection
+    if MiqEnvironment::Command.is_container?
+      host = ENV["ANSIBLE_SERVICE_NAME"]
+      port = 80
+    else
+      host = "localhost"
+      port = HTTP_PORT
+    end
+
     admin_auth = miq_database.ansible_admin_authentication
     AnsibleTowerClient::Connection.new(
-      :base_url => URI::HTTP.build(:host => "localhost", :path => "/api/v1", :port => HTTP_PORT).to_s,
-      :username => admin_auth.userid,
-      :password => admin_auth.password
+      :base_url   => URI::HTTP.build(:host => host, :path => "/api/v1", :port => port).to_s,
```

Comment:
I know this isn't changed in this PR but here and the place we do the URI...build isn't checking if the `host` is nil.  I'm not sure what we can do there but I believe this will blow up building the URI if it's nil.